### DISCUSSION TITLE
Fine looking XML nevertheless is wrong...

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#6167] [Q&A] [answered]
Hi, 

A Response of one of my channels sends out XML, of which these are two examples:

```
<PatientQRY>
   <Patient>
      <IdNumber>22488109069</IdNumber>
      <InternalNumber>29735133</InternalNumber>
      <FirstName>Donna</FirstName>
      <LastName>Winter</LastName>
      <PartnerName />
      <DateOfBirth>1968-05-22</DateOfBirth>
      <PhoneNumber />
   </Patient>
   <Pregnancy>
      <Gravidity />
      <Parity />
      <ExpectedDeliveryDate />
      <FetalNumber />
      <Week />
      <Day />
   </Pregnancy>
   <Error>
      <ErrorMessage />
   </Error>
   <UserName>MessageService</UserName>
</PatientQRY>
```

```
<PatientQRY>
   <Patient>
      <IdNumber>1234567</IdNumber>
      <InternalNumber />
      <FirstName />
      <LastName />
      <PartnerName />
      <DateOfBirth />
      <PhoneNumber />
   </Patient>
   <Pregnancy>
      <Gravidity />
      <Parity />
      <ExpectedDeliveryDate />
      <FetalNumber />
      <Week />
      <Day />
   </Pregnancy>
   <Error>
      <ErrorMessage>No patient found based on national identifier 1234567.</ErrorMessage>
   </Error>
   <UserName>MessageService</UserName>
</PatientQRY>
```

I have obtained these examples by routing the response to a FileWriter channel and getting the output. If I run it through an XML validator, the XML looks fine. Nevertheless, the receiving system states, "There is an error in XML document (0, 0)." This is an issue with one customer of ours, a different customer has a similar channel which works fine. 

Can the XML error be a Mirth Connect issue - wrong encoding perhaps, or something? And if so, how do I resolve it? Mirth Connect 4.4.1 is used,

**Comment by pacmano1** [comment]
```<?xml version="1.0" encoding="UTF-8"?>```

maybe other side expects a proper declration.

**Reply by pacmano1** [reply]
Other stuff - make sure you are not returning something like a BOM (byte order mark) by accident.

**Comment by Coooz** [comment, marked as answer]
It looked like the XML did not arrive at the receiving system at all: by accident I did not specify the response correctly in the Source so my guess is the error message got sent instead. Anyway it is fixed now. Thanks for contributing @pacmano1 .
